cni-plugin/pkg/dataplane/windows/dataplane_windows.go

```diff
@@ -757,6 +757,26 @@ func (d *windowsDataplane) createAndAttachContainerEP(args *skel.CmdArgs,
 		v2pols = append(v2pols, hcnPol)
 	}
 
+	// if supported add loopback DSR
+	//if err := hcn.DSRSupported(); err == nil {
```

Comment:
This doesn't work since the fork of hcsshim being used doesn't have the correct values: https://github.com/projectcalico/hcsshim/blob/v0.8.9-calico/hcn/hcnglobals.go#L41

DSR is not supported in 2019 for over a year. Do you think we should add this check?

Reply:
@jsturtevant Thanks for your PR. Looking good.

> DSR is not supported in 2019 for over a year

Do you mean DSR **is supported** for 2019 for over a year? My understanding is if you get Windows Server 1809 (build 17763.1432 or greater), DSR is supported.

Reply:
Sorry wasn't clear, I meant that it came in the WS 2019 with KB4580390 (Oct 2020) which has been out for a year so I would hope most folks have something newer these days and there for would have DSR support.  I am happy to add the appropriate check, but wasn't sure if it was required? It would require an update to the fork. What do you think?  

But I think I miss understood Calico's DSR requirements.  Reading up more, sounds like DSR is only required if using policy?  from https://projectcalico.docs.tigera.io/getting-started/windows-calico/kubernetes/standard

> For Calico policy to function correctly with Kubernetes services, the WinDSR feature gate must be enabled. This requires Windows Server build 17763.1432 or greater and Kubernetes v1.14 or greater. Calico will automatically enable the WinDSR feature gate if kubernetes services are managed by Calico for Windows.

So this means it is possible someone isn't using calico policy they might not have DSR enabled.  So this should probably be behind a configuration flag?

Reply:
I'm not sure what the implications are of adding the loopback DSR policy if DSR is not enabled. I see some mentions elsewhere like [here](https://github.com/containernetworking/plugins/pull/613#issuecomment-815399853) that suggest that we should check if the host supports DSR.

One option is to update our fork (if we still need it?) to a more recent hcsshim so we could use `hcn.DSRSupported()`. Our fork is over a year behind now so the update might come with other required changes.

We could also add a new field to our CNI [config](https://github.com/projectcalico/calico/blob/master/node/windows-packaging/CalicoWindows/cni.conf.template) like how the [win-overlay/win-bridge plugins](https://github.com/containernetworking/plugins/blob/5238c13aa96ccacb1329903935c83a424e2cd8b6/plugins/main/windows/win-bridge/sample-v2.conf#L50) do it. We already do some [detection](https://github.com/projectcalico/calico/blob/master/node/windows-packaging/CalicoWindows/kubernetes/kube-proxy-service.ps1#L49-L56) of the host Windows OS version to determine if we should enable WinDSR in kube-proxy. Maybe we can pull that up into the calico [lib](https://github.com/projectcalico/calico/blob/master/node/windows-packaging/CalicoWindows/libs/calico/calico.psm1) so that we can also use it to populate `loopbackDSR` in the Calico CNI config.

What do you guys think? I think the latter option would allow us to backport this fix more easily.

Reply:
I think we should support Kubernetes networking without DSR support. We should check if DSR is supported before applying loopback policies. Calico users may still have some old OS version running. 

@lmm Agree adding new field to CNI config is a better option right now. Making backport easier and it also more flexible since one can always overwrite the config if they need to. 

@jsturtevant Maybe you can update your PR to keep `hcn.DSRSupported()` and just adding loopback policy support. We can raise a follow-up PR to replace it with a config variable.  WDYT?

Reply:
We need to update the fork before `hcn.DSRSupported()` will work.  I like the idea of using the powershell to set those variables properly during installation with a config variable.

Otherwise I think the fork needs to be updated, which we could also do in a separate PR.